service/src/main/java/bio/terra/workspace/service/workspace/AzureCloudContextService.java

```diff
@@ -111,11 +111,16 @@ public CloudContext makeCloudContextFromDb(DbCloudContext dbCloudContext) {
   public List<ControlledResource> makeOrderedResourceList(UUID workspaceUuid) {
     List<ControlledResource> unorderedList =
         resourceDao.listControlledResources(workspaceUuid, CloudPlatform.AZURE);
-    // Delete VMs first because they use other resources like disks, networks, etc.
+    // Delete VMs and k8s namespaces first because they use other resources like disks, identities,
+    // databases, etc.
     List<ControlledResource> orderedList =
         new ArrayList<>(
             unorderedList.stream()
-                .filter(r -> r.getResourceType() == WsmResourceType.CONTROLLED_AZURE_VM)
+                .filter(
+                    r ->
+                        r.getResourceType() == WsmResourceType.CONTROLLED_AZURE_VM
```

Comment:
What about line 138? Maybe there is a way to rewrite this code so that the "special" resources don't have to be listed in multiple places.